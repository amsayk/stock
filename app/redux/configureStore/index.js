// @flow

import { createStore, applyMiddleware, compose } from 'redux';

import { APP_NAME, DEBUG, INIT } from 'vars';

import pick from 'lodash.pick';

import array from 'redux/middleware/array';
import bufferActions from 'redux/middleware/buffer-actions';

import { createHashHistory } from 'history';
import { routerMiddleware, routerActions } from 'react-router-redux';
import locationHelperBuilder from 'redux-auth-wrapper/history4/locationHelper';

import { client as apolloClient } from 'apollo';

import { createSnackbarController } from 'components/Snackbar';

import thunk from 'redux-thunk';

import rootReducer from 'redux/reducers';

const history = createHashHistory();

const configureStore = (initialState?: any) => {
  // ======================================================
  // Middleware Configuration
  // ======================================================
  const middlewares = [
    bufferActions(),
    array,
    thunk.withExtraArgument({
      client: apolloClient,
      history,
      get snackbar() {
        return (
          store.snackbar || (store.snackbar = createSnackbarController(store))
        );
      },
    }),
    routerMiddleware(history),
  ];

  // Redux DevTools Configuration
  const actionCreators = {
    ...routerActions,
  };

  const composeEnhancers =
    (__DEV__ || DEBUG) && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
      ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
          name: APP_NAME,
          routerActions,
        })
      : compose;

  // ======================================================
  // Store Enhancers
  // ======================================================
  const enhancers = [applyMiddleware(...middlewares)];

  const enhancer = composeEnhancers(...enhancers);

  const store = createStore(rootReducer, enhancer);

  // Snackbar
  store.snackbar = createSnackbarController(store);

  // Initialize
  store.dispatch({ type: INIT });

  if (__DEV__) {
    if (module.hot) {
      module.hot.accept('../reducers', () => {
        store.replaceReducer(require('../reducers'));
      });
    }
  }

  return store;
};

export default {
  configureStore,
  history,
  locationHelper: locationHelperBuilder({}),
};
