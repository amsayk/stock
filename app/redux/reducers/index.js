import user from './user/reducer';
import app from './app/reducer';
import snackbar from './snackbar/reducer';
import notification from './notification/reducer';
import scrolling from './scrolling/reducer';
import toastr from './toastr/reducer';
import intl from './intl/reducer';

import { USER_LOGGED_OUT } from './user/constants';

import { combineReducers } from 'redux-immutable';

import { INIT } from 'vars';

import { routerReducer as router } from 'react-router-redux';

import { reducer as formReducer, actionTypes } from 'redux-form/immutable';

const reducers = {
  intl,
  app,
  notification,
  scrolling,
  user,
  snackbar,
  toastr,
  form: formReducer.plugin({
    login: (state, action) => {
      switch (action.type) {
        case USER_LOGGED_OUT:
          return state
            ? state.mergeDeep({
                values: {
                  password: undefined,
                },
              })
            : state;
        case actionTypes.SUBMIT_START: // Clear errors
          return state
            ? state.merge({
                error: null,
              })
            : state;
        default:
          return state;
      }
    },
    changePassword: (state, action) => {
      switch (action.type) {
        case actionTypes.SET_SUBMIT_SUCCEEDED: // Clear passwords on success
          return state
            ? state.merge({
                values: {},
              })
            : state;
        default:
          return state;
      }
    },
    changeEmail: (state, action) => {
      switch (action.type) {
        case actionTypes.SET_SUBMIT_SUCCEEDED: // Clear email on success
          return state
            ? state.merge({
                values: {},
              })
            : state;
        default:
          return state;
      }
    },
  }),
};

export default combineReducers({
  router,
  ...reducers,
});
