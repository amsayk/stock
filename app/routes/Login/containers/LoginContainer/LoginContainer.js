import React from 'react';
import T from 'prop-types';

import { locationHelper } from 'redux/configureStore';
import Link from 'react-router-dom/Link';

import { compose, bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import checkBusiness from 'utils/checkBusiness';

import { logIn } from 'redux/reducers/user/actions';

import { createSelector } from 'utils/reselect';

import style from 'routes/Login/styles';

import Title from 'components/Title';

import isEmpty from 'isEmpty';

import {
  SubmissionError,
  Field,
  reduxForm,
  propTypes as formPropTypes,
} from 'redux-form/immutable';

import AppLogo from 'components/AppLogo';

import { intlShape, injectIntl } from 'react-intl';

import messages from 'routes/Login/messages';

import {
  APP_NAME,
  LINK_PRIVACY_POLICY,
  LINK_TERMS_OF_SERVICE,
  LINK_SUPPORT,
} from 'vars';

export class LoginContainer extends React.Component {
  static propTypes = {
    ...formPropTypes,
    intl: intlShape.isRequired,
    actions: T.shape({
      logIn: T.func.isRequired,
    }).isRequired,
  };

  constructor(props) {
    super(props);

    this.onSubmit = this.onSubmit.bind(this);
    this.onKeyDown = this._onKeyDown.bind(this);
  }

  _onKeyDown(e) {
    if (e.key === 'Enter' && e.shiftKey === false) {
      const submit = this.props.handleSubmit(this.onSubmit);
      submit();
    }
  }

  async onSubmit(data) {
    const { intl, actions } = this.props;
    const { username, password } = data.toJS();

    try {
      await actions.logIn(
        username,
        /*password = */ isEmpty(password) && __DEV__
          ? process.env.DEV_PASSWORD
          : password,
      );

      if (__DEV__) {
        localStorage.setItem('app.logIn', username);
      }

      // async: Notify busines info
      checkBusiness();
    } catch (e) {
      throw new SubmissionError({ _error: intl.formatMessage(messages.error) });
    }
  }

  _renderForm() {
    const { intl, error, handleSubmit, submitting } = this.props;

    return [
      <h1 className={style.heading}>
        {intl.formatMessage(messages.title, { appName: APP_NAME })}
      </h1>,

      <Field
        name="username"
        component={'input'}
        type="text"
        placeholder={intl.formatMessage(messages.username)}
        className={style.usernameFieldControl}
        autoComplete="off"
        onKeyDown={this.onKeyDown}
        autoFocus
      />,

      <Field
        name="password"
        component="input"
        type="password"
        placeholder={intl.formatMessage(messages.password)}
        className={style.passwordFieldControl}
        onKeyDown={this.onKeyDown}
      />,

      <div className={style.error}>
        {error && !submitting ? (
          <small className={style.formControlFeedback}>
            {intl.formatMessage(messages.error)}
          </small>
        ) : null}
      </div>,

      <button
        onClick={handleSubmit(this.onSubmit)}
        disabled={submitting}
        className={style.logIn}
        role="button"
      >
        {intl.formatMessage(messages.logIn)}
      </button>,

      <br />,
    ];
  }

  render() {
    const { intl } = this.props;
    return (
      <div className={style.root}>
        <Title
          title={intl.formatMessage(messages.pageTitle, { appName: APP_NAME })}
        />
        <div className={style.center}>
          <Link className={style.logo} to="/">
            <AppLogo width={52} height={52} />
          </Link>
          <div className={style.form}>{this._renderForm()}</div>
        </div>
        {/* <footer> */}
        {/*   <ul> */}
        {/*     <li className={style.footerLink}> */}
        {/*       <a target="_blank" href={LINK_TERMS_OF_SERVICE}> */}
        {/*         {intl.formatMessage(messages.termsOfService)} */}
        {/*       </a> */}
        {/*     </li> */}
        {/*     <li className={style.footerLink}> */}
        {/*       <a target="_blank" href={LINK_SUPPORT}> */}
        {/*         {intl.formatMessage(messages.support)} */}
        {/*       </a> */}
        {/*     </li> */}
        {/*     <li className={style.footerLink}> */}
        {/*       <a target="_blank" href={LINK_PRIVACY_POLICY}> */}
        {/*         {intl.formatMessage(messages.privacyPolicy)} */}
        {/*       </a> */}
        {/*     </li> */}
        {/*   </ul> */}
        {/* </footer> */}
      </div>
    );
  }
}

const redirectSelector = (_, ownProps) =>
  locationHelper.getRedirectQueryParam(ownProps);
const initialValuesSelector = () => ({
  username: __DEV__ ? localStorage.getItem('app.logIn') : null,
});

const mapStateToProps = createSelector(
  redirectSelector,
  initialValuesSelector,
  (redirect, initialValues) => ({ redirect, initialValues }),
);

function mapDispatchToProps(dispatch) {
  return { actions: bindActionCreators({ logIn }, dispatch) };
}

const Connect = connect(mapStateToProps, mapDispatchToProps);

const WithForm = reduxForm({
  form: 'logIn',
});

export default compose(injectIntl, Connect, WithForm)(LoginContainer);
