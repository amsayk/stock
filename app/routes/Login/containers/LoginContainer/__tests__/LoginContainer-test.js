import React from 'react';
import { compose } from 'redux';
import configureMockStore from 'redux-mock-store';
import { reduxForm } from 'redux-form/immutable';
import { IntlProvider, injectIntl } from 'react-intl';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';

import { fromJS } from 'immutable';

import { LoginContainer } from '../LoginContainer';

const Decorated = compose(injectIntl, reduxForm({ form: 'testForm' }))(
  LoginContainer,
);

const middlewares = [];
const mockStore = configureMockStore(middlewares);

describe('login container', () => {
  const store = mockStore(
    fromJS({
      form: {
        testForm: {},
      },
    }),
  );

  it('should render the snapshot', () => {
    const tree = renderer
      .create(
        <IntlProvider
          defaultLocale={'en'}
          locale={'en'}
          messages={{}}
          formats={{}}
        >
          <Provider store={store}>
            <Decorated
              actions={{
                login: jest.fn(),
              }}
            />
          </Provider>
        </IntlProvider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
