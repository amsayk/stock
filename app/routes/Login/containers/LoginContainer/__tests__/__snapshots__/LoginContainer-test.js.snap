exports[`login container should render the snapshot 1`] = `
<div
  className="root">
  <div
    className="center">
    <a
      className="logo"
      onClick={[Function]}
      style={Object {}}>
      <svg
        baseProfile="tiny"
        className="logo"
        height="52px"
        version="1.1"
        viewBox="0 0 179 179"
        width="52px"
        xmlns="http://www.w3.org/2000/svg">
        <path
          d="M36.57,116.471c-1.184,1.211-2.773,1.877-4.483,1.877c-1.652,0-3.204-0.632-4.383-1.784l-0.11-0.108 c-1.194-1.162-1.863-2.725-1.879-4.398c-0.026-1.679,0.61-3.26,1.773-4.451l12.008-12.188c1.184-1.209,2.778-1.875,4.482-1.875 c1.652,0,3.21,0.631,4.373,1.768l0.121,0.121c1.194,1.166,1.862,2.734,1.878,4.41c0.021,1.673-0.61,3.248-1.772,4.438L36.57,116.471 z M52.361,99.82c-0.026-2.213-0.91-4.281-2.479-5.816l-0.121-0.121c-1.553-1.514-3.604-2.348-5.783-2.348 c-2.246,0-4.352,0.882-5.914,2.477L26.057,106.2c-1.541,1.579-2.378,3.665-2.352,5.878c0.026,2.212,0.91,4.275,2.494,5.822 l0.11,0.104c1.547,1.513,3.6,2.35,5.777,2.35c2.252,0,4.352-0.882,5.914-2.479l12.008-12.191 C51.551,104.111,52.382,102.03,52.361,99.82"
          fill="currentColor" />
        <path
          d="M51.324,127.348c-1.184,1.207-2.772,1.873-4.478,1.873c-1.657,0-3.21-0.631-4.389-1.787l-0.11-0.102 c-2.468-2.418-2.51-6.389-0.105-8.851l12.003-12.194c1.188-1.207,2.783-1.873,4.488-1.873c1.646,0,3.204,0.629,4.378,1.775 l0.115,0.113c1.194,1.166,1.857,2.73,1.879,4.406c0.021,1.674-0.61,3.252-1.773,4.441L51.324,127.348z M67.115,110.686 c-0.031-2.213-0.915-4.277-2.489-5.816l-0.115-0.116c-1.547-1.513-3.604-2.347-5.777-2.347c-2.252,0-4.352,0.881-5.92,2.476 l-12.002,12.194c-3.184,3.258-3.126,8.5,0.142,11.697l0.11,0.105c1.547,1.513,3.604,2.347,5.783,2.347 c2.252,0,4.352-0.879,5.909-2.474l12.007-12.196C66.305,114.982,67.142,112.893,67.115,110.686"
          fill="currentColor" />
        <path
          d="M67.025,136.922c-1.184,1.21-2.772,1.875-4.482,1.875c-1.652,0-3.205-0.631-4.373-1.768l-0.121-0.121 c-1.188-1.166-1.857-2.729-1.878-4.404c-0.021-1.676,0.61-3.254,1.773-4.446l12.013-12.191c1.179-1.208,2.768-1.876,4.472-1.876 c1.658,0,3.216,0.629,4.384,1.775l0.115,0.113c1.194,1.166,1.863,2.731,1.879,4.407c0.021,1.676-0.61,3.255-1.773,4.44 L67.025,136.922z M82.816,120.262c-0.026-2.209-0.91-4.274-2.483-5.818l-0.121-0.113c-1.542-1.514-3.599-2.345-5.783-2.345 c-2.252,0-4.352,0.882-5.903,2.476l-12.013,12.195c-1.542,1.575-2.379,3.658-2.353,5.871c0.026,2.211,0.911,4.275,2.484,5.81 l0.121,0.118c1.547,1.513,3.599,2.35,5.777,2.35c2.252,0,4.352-0.881,5.914-2.476l12.008-12.194 C82.006,124.559,82.843,122.473,82.816,120.262"
          fill="currentColor" />
        <path
          d="M81.774,147.361c-1.179,1.213-2.772,1.878-4.478,1.878c-1.652,0-3.204-0.629-4.383-1.784l-0.11-0.104 c-2.468-2.415-2.516-6.391-0.105-8.853l12.008-12.189c1.184-1.211,2.772-1.876,4.478-1.876c1.652,0,3.21,0.632,4.372,1.763 l0.127,0.129c1.194,1.166,1.862,2.729,1.878,4.404c0.021,1.673-0.61,3.252-1.773,4.441L81.774,147.361z M97.571,130.707 c-0.026-2.213-0.911-4.277-2.479-5.812l-0.132-0.126c-1.542-1.511-3.599-2.345-5.777-2.345c-2.252,0-4.352,0.882-5.914,2.476 l-12.008,12.191c-3.184,3.258-3.12,8.504,0.147,11.7l0.11,0.108c1.547,1.512,3.604,2.346,5.777,2.346 c2.252,0,4.356-0.881,5.914-2.48l12.008-12.191C96.755,135.001,97.592,132.917,97.571,130.707"
          fill="currentColor" />
        <path
          d="M26.126,103.304L7.499,87.275l23.594-41c8.104,4.699,16.574,6.146,23.957,4.064 c0.815-0.229,1.631-0.463,2.452-0.694c8.098-2.308,16.47-4.688,24.878-4.688c4.525,0,6.403,0.931,7.188,1.713l0.711,0.71l1.42-1.421 l-0.715-0.708c-1.548-1.547-4.362-2.302-8.604-2.302c-8.688,0-17.195,2.423-25.431,4.767c-0.815,0.231-1.626,0.463-2.441,0.692 c-9.593,2.702-18.685-1.581-23.252-4.373l-0.879-0.539L4.931,87.713l19.885,17.108L26.126,103.304z"
          fill="currentColor" />
        <path
          d="M147.706,109.218l-1.974,0.358l0.179,0.983c0.105,0.603,0.979,5.956-1.637,9.147 c-1.347,1.65-3.504,2.484-6.414,2.484c-1.157,0-2.431-0.129-3.793-0.385l-1.921-0.357l-0.185,1.949 c-0.031,0.348-0.868,8.559-9.655,8.559c-1.489,0-3.146-0.231-4.914-0.691l-2.132-0.553l-0.1,2.207 c-0.016,0.332-0.442,8.156-8.756,8.156c-0.947,0-1.979-0.101-3.067-0.293l-1.505-0.271l-0.489,1.434 c-0.116,0.323-2.895,7.866-8.935,7.866c-1.542,0-3.231-0.48-5.015-1.434l-0.884-0.471l-0.947,1.771l0.889,0.474 c2.079,1.107,4.084,1.668,5.957,1.668c6.945,0,10.239-7.695,10.76-9.031c1.137,0.193,2.227,0.294,3.236,0.294 c9.397,0,10.592-8.183,10.739-9.811c1.842,0.458,3.573,0.692,5.162,0.692c8.293,0,11.202-6.502,11.629-10.143 c1.395,0.25,2.715,0.376,3.925,0.376c3.536,0,6.22-1.084,7.967-3.22c3.12-3.805,2.263-9.643,2.058-10.771L147.706,109.218z"
          fill="currentColor" />
        <path
          d="M154.599,105.25c-0.026,0.021-0.047,0.043-0.068,0.063l-0.084,0.081c-0.016,0.016-0.405,0.398-1.179,0.398 c-0.806,0-1.778-0.406-2.815-1.177c-1.726-2.013-24.525-28.369-42.615-37.246l-0.469-0.229c-0.368-0.182-0.8-0.121-1.104,0.147 l-0.395,0.35c-1.595,1.418-3.268,2.849-5.104,4.38c-3.483,2.897-7.63,6.256-11.965,9.196c-1.652,1.12-3.432,2.283-5.373,3.064 c-3.436,1.384-7.119,1.758-10.045,0.989c-1.115-0.292-1.937-0.942-2.983-1.966c-2.058-2.015-4.193-5.08-2.878-8.35 c2.463-1.863,16.97-13.057,32.523-29.727c0.121-0.105,1.405-1.129,4.473-1.129c3.567,0,8.045,1.3,13.307,3.873 c0.021,0.008,0.037,0.016,0.059,0.024l0.089,0.04c0.021,0.008,0.037,0.013,0.053,0.019c0.09,0.031,9.13,3.17,16.417,3.886 c0.348,0.042,1.315,0.145,2.763,0.145c3.999,0,11.681-0.805,18.843-6.117c2.157,4.078,11.008,28.49,16.548,44.257L154.599,105.25z M174.72,90.215c-0.042-0.119-4.072-11.597-8.219-22.931c-8.613-23.576-9.161-23.576-10.145-23.576 c-0.174,0-0.321,0.005-0.427,0.016c-0.194,0.021-0.373,0.089-0.526,0.208c-6.855,5.346-14.349,6.146-18.2,6.146 c-1.368,0-2.268-0.103-2.542-0.134c-6.893-0.676-15.428-3.594-15.948-3.775l-0.037-0.016c-5.524-2.697-10.292-4.062-14.159-4.062 c-3.968,0-5.631,1.46-5.914,1.744l-0.006-0.002C82.606,60.98,67.747,72.277,66.1,73.508c-0.132,0.103-0.242,0.237-0.311,0.389 c-1.552,3.444-0.415,7.293,3.205,10.84c1.025,1.005,2.173,2.025,3.878,2.473c3.346,0.879,7.472,0.474,11.302-1.071 c2.126-0.854,4.005-2.08,5.752-3.264c4.403-2.984,8.598-6.386,12.117-9.314c1.821-1.51,3.468-2.928,5.047-4.323 C125.111,78.157,148.026,104.866,149,106c0.042,0.059,0.095,0.105,0.152,0.147c1.442,1.097,2.826,1.652,4.115,1.652 c1.199,0,2.136-0.513,2.609-0.995l18.538-15.49C174.735,91.049,174.856,90.606,174.72,90.215"
          fill="currentColor" />
      </svg>
    </a>
    <div
      className="form">
      <h1
        className="heading">
        Log In to 
      </h1>
      <input
        autoComplete="off"
        autoFocus={true}
        className="emailFieldControl"
        name="email"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="Email"
        type="text"
        value="" />
      <input
        className="passwordFieldControl"
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="Password"
        type="password"
        value="" />
      <div
        className="error" />
      <button
        className="logIn"
        disabled={false}
        onClick={[Function]}>
        Log In
      </button>
      <div
        className="passwordReset">
        <a
          className="passwordResetButton"
          onClick={[Function]}
          style={Object {}}>
          Forgot your password?
        </a>
      </div>
    </div>
    <div
      className="navSignupWrapper">
      <span
        className="navbarText">
        Not a member yet?
      </span>
       
      <a
        className="joinButton"
        onClick={[Function]}
        style={Object {}}>
        Join now!
      </a>
    </div>
  </div>
  <footer />
</div>
`;
